lib/src/source_visitor.dart

```diff
@@ -761,8 +761,11 @@ class SourceVisitor extends ThrowingAstVisitor {
 
     var needsDouble = true;
     for (var declaration in node.declarations) {
-      // Add a blank line before classes.
-      if (declaration is ClassDeclaration) needsDouble = true;
+      var hasBody = declaration is ClassDeclaration ||
```

Comment:
[nit] Is there any more specific name we can give this? It feels a little jarring to see us pull out a `body` variable in the condition for `!hasBody` below...